include/aspect/material_model/drucker_prager.h

```diff
@@ -0,0 +1,248 @@
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef __aspect__model_drucker_prager_h
+#define __aspect__model_drucker_prager_h
+
+#include <aspect/material_model/interface.h>
+#include <aspect/simulator_access.h>
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    using namespace dealii;
+
+    /**
+     * A material model that consists of globally constant values for all
+     * material parameters except density and viscosity.
+     *
+     * The model is considered incompressible, following the definition
+     * described in Interface::is_compressible.
+     *
+     * The viscosity is computed according to the Drucker Prager frictional
+     * plasticity criterion based on a user-defined Internal angle of friction $\\phi$
+     * and cohesion $C$. In 3D:
+     * $\\sigma_y = \\frac{6 C cos(\\phi)}{\\sqrt(3) (3+sin(\\phi))} +\\
+     * \\frac{2 P sin(\\phi)}{\\sqrt(3) (3+sin(\\phi))}$,
+     * where P is the pressure.
+     * See for example Zienkiewicz, O. C., Humpheson, C. and Lewis, R. W. (1975),
+     * G\'{e}otechnique 25, No. 4, 671-689.
+     * With this formulation we circumscribe instead of inscribe the Mohr Coulomb
+     * yield surface.
+     * In 2D the Drucker Prager yield surface is the same
+     * as the Mohr Coulomb surface:
+     * $\\sigma_y = P sin(\\phi) + C cos(\\phi)$.
+     * Note that in 2D for $\\phi=0$, these criteria
+     * revert to the von Mises criterion (no pressure dependence).
+     * See for example Thieulot, C. (2011), PEPI 188, 47-68.
+     *
+     * Note that we enforce the pressure to be positive to prevent negative
+     * yield strengths and viscosities.
+     * We then use the computed yield strength to scale back the viscosity on
+     * to the yield surface using the Viscosity Rescaling Method described in
+     * Kachanov, L. M. (2004), Fundamentals of the Theory of Plasticity,
+     * Dover Publications, Inc.
+     *
+     * To avoid numerically unfavourably large (or even negative) viscosity ranges,
+     * we cut off the viscosity with a user-defined minimum and maximum viscosity:
+     * $\\eta_eff = \\frac{1}{\\frac{1}{max(\\eta_min,\\eta)}+\\
+     * \\frac{1}{\\eta_max}}.
+     *
+     * Note that this model uses the formulation that assumes an incompressible
+     * medium despite the fact that the density follows the law
+     * $\\rho(T)=\\rho_0(1-\\beta(T-T_{\\text{ref}}))$.
+     *
+     *
+     * @ingroup MaterialModels
+     */
+    template <int dim>
+    class DP : public MaterialModel::InterfaceCompatibility<dim>, public ::aspect::SimulatorAccess<dim>
```

Comment:
I missed this the first time around -- would you be willing to spell the class `DruckerPrager` to make it more obvious to people like me who don't know the abbreviation?